### PR TITLE
Andrew Ghobrial --- Testing

### PR DESCRIPTION
**Pull request by ghobs91**
@gooseandmegander

### CODE REVIEW

File: basic-javascript/project-1.spec.js [hunk 1]
```diff
@@ -1,3 +1,173 @@
 const helpers = require('./project-1');
 
 // start testing!
+
+describe('project-1.js', () => {
+ 
+  // multiplyByTen
+  test("should provide correct value when multiplying by 10", () => {
+    const expected = 50;
+    const actual = helpers.multiplyByTen(5);
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:10]: What if a string is supplied? Or nothing?

File: basic-javascript/project-1.spec.js [hunk 2]
```diff
@@ -1,3 +1,173 @@
...
+    expect(actual).toEqual(expected);
+  });
+
+  // subtractFive
+  test("should provide correct value when subtracting by 5", () => {
+    const expected = 20;
+    const actual = helpers.subtractFive(25);
+    expect(actual).toEqual(expected);
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:19]: Depending on the context, should the output be allowed to be negative?

File: basic-javascript/project-1.spec.js [hunk 3]
```diff
@@ -1,3 +1,173 @@
...
+
+  // areSameLength
+  test("should return true when both inputs are in fact the same length", () => {
+    const expected = true;
+    const actual = helpers.areSameLength("car", "and");
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:24]: What if a boolean is supplied?

File: basic-javascript/project-1.spec.js [hunk 4]
```diff
@@ -1,3 +1,173 @@
...
+    expect(actual).toEqual(expected);
+  });
+
+  // areEqual
+  test("should return true when both inputs do in fact have deep equality", () => {
+    const expected = true;
+    const actual = helpers.areEqual(2, 2);
+    expect(actual).toEqual(expected);
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:33]: For deep equality, I might also check an object or array that's nested within another object/array.

File: basic-javascript/project-1.spec.js [hunk 5]
```diff
@@ -1,3 +1,173 @@
...
+    const actual = helpers.subtract(100, 30);
+    expect(actual).toEqual(expected);
+  });
+
+  // divide
+  test("should return correct quotient of two numbers", () => {
+    const expected = 25;
+    const actual = helpers.divide(100, 4);
+    expect(actual).toEqual(expected);
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:68]: What should happen if the denominator is zero?

File: basic-javascript/project-1.spec.js [hunk 6]
```diff
@@ -1,3 +1,173 @@
...
+  // roundNumber
+  test("should correctly round a number", () => {
+    const actual = helpers.roundNumber(5.6);
+    expect(actual).toBe(6);
+  });
+
+  // addExclamationPoint
+  test("should successfully append an exclamation point to a string", () => {
+    expect(helpers.addExclamationPoint("I went to the store")).toBe("I went to the store!");
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:137]: What if there's a number supplied?

File: basic-javascript/project-1.spec.js [hunk 7]
```diff
@@ -1,3 +1,173 @@
...
+
+  // combineNames
+  test("should successfully combine two names inputted into a string with both", () => {
+    expect(helpers.combineNames("Andrew", "Ghobrial")).toBe("Andrew Ghobrial");
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-1.spec.js:142]: What if an input is missing?

File: basic-javascript/project-2.spec.js [hunk 8]
```diff
@@ -1,3 +1,144 @@
...
 // whoops.. there is no test suite for this file. You'll simply just have to create one :/
+
+describe('project-2.js', () => { 
+
+  // getBiggest
+  test("should return the larger input", () => {
+    const expected = 8;
+    const actual = helpers.getBiggest(5, 8);
+    expect(actual).toEqual(expected);
+  });
```
Comment (gooseandmegander) [inline basic-javascript/project-2.spec.js:12]: What if a string is provided?

**Review by gooseandmegander — COMMENTED** [review]
# Objectives
* Familiarity with writing unit tests for JavaScript code.

# MVP
- [x] project-1.spec.js tests passing
- [x] project-2.spec.js tests passing

# Stretch
- [ ] arrays.spec.js tests passing
- [ ] objects.spec.js tests passing

# Great
All the tests pass. There are "green" tests.

# Requests for Improvement
* Write some "red" tests for cases that shouldn't pass. How will different primitive values affect the output? How does the context of the test affect what the inputs _ought_ to be? Etc.
* Check out other types of testing.
* Use something new from the docs you haven't used yet to generate tests.
* Think about edge cases.

# Rating: 2
